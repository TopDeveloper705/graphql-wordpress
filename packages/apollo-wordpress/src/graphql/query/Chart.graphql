query Chart {
  viewer {
    chart {
      title
      copyright
      updated
      authorName
      authorUri
      items {
        title
        artist
        releaseDate
        releaseDateFormatted
        url
        copyright
        images {
          url
          height
        }
      }
    }
  }
}
