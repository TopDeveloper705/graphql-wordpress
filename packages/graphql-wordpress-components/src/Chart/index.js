import styled from 'react-emotion';
import { clear } from '../global';

export const Title = styled.h1`
  font-family: ${p => p.theme.fonts.futura};
  font-size: 36px;
  font-weight: bold;
  line-height: 42px;
  margin: 0 0 10px;

  & a {
    color: ${p => p.theme.colors.dark};
    text-decoration: none;
  }
`;

export const List = styled.ol`list-style: decimal;`;

export const Item = styled.li`
  ${clear};
  display: list-item;
  margin: 10px 0 10px 20px;
  padding: 0 0 0 7px;
`;

export const Image = styled.img`
  display: block;
  float: left;
  margin: 0 10px 0 0;
`;
