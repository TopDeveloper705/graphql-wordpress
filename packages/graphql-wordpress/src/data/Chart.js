import { toGlobalId } from 'graphql-relay';

class Chart {
  getID() {
    return toGlobalId(this.constructor.name, 'me');
  }

  static getEndpoint() {
    return 'https://itunes.apple.com/us/rss/topalbums/limit=25/explicit=true/json';
  }
}

export default Chart;
