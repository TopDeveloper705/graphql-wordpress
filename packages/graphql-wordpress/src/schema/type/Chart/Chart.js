import { GraphQLObjectType, GraphQLString, GraphQLList } from 'graphql';
import ChartItemType from './ChartItem';

const ChartType = new GraphQLObjectType({
  name: 'Chart',
  description: 'iTunes chart data.',
  fields: {
    title: {
      type: GraphQLString,
      description: 'Chart title.',
      resolve: chart => chart.title.label,
    },
    copyright: {
      type: GraphQLString,
      description: 'Chart copyright notice.',
      resolve: chart => chart.rights.label,
    },
    updated: {
      type: GraphQLString,
      description: 'Last updated timestamp.',
      resolve: chart => chart.updated.label,
    },
    authorName: {
      type: GraphQLString,
      description: 'Chart author name.',
      resolve: chart => chart.author.name.label,
    },
    authorUri: {
      type: GraphQLString,
      description: 'Chart author URI.',
      resolve: chart => chart.author.uri.label,
    },
    items: {
      type: new GraphQLList(ChartItemType),
      description: 'Chart items.',
      resolve: chart => chart.entry,
    },
  },
});

export default ChartType;
