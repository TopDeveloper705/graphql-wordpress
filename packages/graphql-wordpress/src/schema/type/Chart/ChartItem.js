import { GraphQLObjectType, GraphQLString, GraphQLList } from 'graphql';
import ChartItemImageType from './ChartItemImage';

const ChartItemType = new GraphQLObjectType({
  name: 'ChartItem',
  description: 'An item in the iTunes chart.',
  fields: {
    title: {
      type: GraphQLString,
      description: 'Album title.',
      resolve: item => item['im:name'].label,
    },
    artist: {
      type: GraphQLString,
      description: 'Artist title.',
      resolve: item => item['im:artist'].label,
    },
    releaseDate: {
      type: GraphQLString,
      description: 'Album release date in ISO format.',
      resolve: item => item['im:releaseDate'].label,
    },
    releaseDateFormatted: {
      type: GraphQLString,
      description: 'Album release date, formatted for display.',
      resolve: item => item['im:releaseDate'].attributes.label,
    },
    url: {
      type: GraphQLString,
      description: 'Album URL.',
      resolve: item => item.link.attributes.href,
    },
    copyright: {
      type: GraphQLString,
      description: 'Chart copyright notice.',
      resolve: item => item.rights.label,
    },
    images: {
      type: new GraphQLList(ChartItemImageType),
      description: 'Item images.',
      resolve: item => item['im:image'],
    },
  },
});

export default ChartItemType;
