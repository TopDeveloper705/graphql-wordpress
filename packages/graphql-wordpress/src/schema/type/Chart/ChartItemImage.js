import { GraphQLObjectType, GraphQLString, GraphQLInt } from 'graphql';

const ChartItemImageType = new GraphQLObjectType({
  name: 'ChartItemImage',
  description: 'iTunes chart item image.',
  fields: {
    url: {
      type: GraphQLString,
      description: 'Image URL.',
      resolve: image => image.label,
    },
    height: {
      type: GraphQLInt,
      description: 'Image height.',
      resolve: image => image.attributes.height,
    },
  },
});

export default ChartItemImageType;
