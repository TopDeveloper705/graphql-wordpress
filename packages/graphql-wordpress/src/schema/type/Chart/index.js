import ChartType from './Chart';

export default ChartType;
