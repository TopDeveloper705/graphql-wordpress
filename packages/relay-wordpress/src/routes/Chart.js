// @flow
import React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { ArticleWrapper, Error } from '@wonderboymusic/graphql-wordpress-components';
import { Title, List, Item, Image } from '@wonderboymusic/graphql-wordpress-components/lib/Chart';

type ChartImage = {|
  url: string,
  height: number,
|};

type ChartItem = {|
  title: string,
  artist: string,
  releaseDate: string,
  releaseDateFormatted: string,
  url: string,
  copyright: string,
  images: Array<ChartImage>,
|};

type ITunesChart = {|
  title: string,
  copyright: string,
  updated: string,
  authorName: string,
  authorUri: string,
  items: Array<ChartItem>,
|};

type ChartProps = {
  viewer: {|
    chart: ITunesChart,
  |},
};

const Chart = ({ viewer: { chart } }: ChartProps) => {
  if (!chart) {
    return <Error />;
  }

  return (
    <ArticleWrapper>
      <header>
        <Title>
          <a href={chart.authorUri}>{chart.authorName}</a>
        </Title>
      </header>
      <List>
        {chart.items.map(({ title, url, artist, releaseDateFormatted, images }) => (
          <Item key={url}>
            {images.length && <Image src={images[0].url} alt="" />}
            <a href={url} target="_blank" rel="noopener noreferrer">
              {title} - {artist}
            </a>
            <p>
              <strong>Released:</strong> {releaseDateFormatted}
            </p>
          </Item>
        ))}
      </List>
    </ArticleWrapper>
  );
};

export default createFragmentContainer(
  Chart,
  graphql`
    fragment Chart_viewer on Viewer {
      chart {
        title
        copyright
        updated
        authorName
        authorUri
        items {
          title
          artist
          releaseDate
          releaseDateFormatted
          url
          copyright
          images {
            url
            height
          }
        }
      }
    }
  `
);
