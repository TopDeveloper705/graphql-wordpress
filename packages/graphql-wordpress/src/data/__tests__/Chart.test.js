import Chart from 'data/Chart';

describe('Test Chart data access', () => {
  test('Get endpoint', () => {
    expect(Chart.getEndpoint()).toMatchSnapshot();
  });
});
