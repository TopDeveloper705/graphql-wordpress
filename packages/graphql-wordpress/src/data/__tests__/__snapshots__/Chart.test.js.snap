// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Chart data access Get endpoint 1`] = `"https://itunes.apple.com/us/rss/topalbums/limit=25/explicit=true/json"`;
